src/ui/ui_nav_link.js

```diff
@@ -3,6 +3,7 @@ import { join } from 'path';
 
 export default class UiNavLink {
   constructor(uiExports, spec) {
+    this.id = spec.id;
```

Comment:
Since there are other things that rely on this property, can you add a test to verify that the constructor sets it properly?